### PR TITLE
Revive logging in Plank

### LINKED ISSUE #14121: Add verbose logging for plank CRUD operations
**What should be cleaned up or changed**:
Now that `plank` is using the new k8s client (#13980) logging is no longer being performed on the various *prowjob* and *pod* CRUD operations (e.g. GetProwJob, ReplaceProwJob, etc.). Logging should be added when these operations are performed (e.g. https://github.com/kubernetes/test-infra/blob/master/prow/plank/controller.go#L378).

**Provide any links for context**:
https://github.com/kubernetes/test-infra/blob/master/prow/kube/client.go#L418
https://github.com/kubernetes/test-infra/blob/master/prow/kube/client.go#L443
https://github.com/kubernetes/test-infra/blob/master/prow/kube/client.go#L474

### PR DESCRIPTION
**Pull request by clarketm**
Revive logging in `plank` for client CRUD operations on ProwJobs / Pods. 

image

resolved #14121

/assign @stevekuznetsov @cjwagner

### CODE REVIEW

File: prow/plank/controller.go [hunk 1]
```diff
@@ -375,7 +380,10 @@ func (c *Controller) syncPendingJob(pj prowapi.ProwJob, pm map[string]coreapi.Po
 			if !ok {
 				return fmt.Errorf("unknown pod %s: unknown cluster alias %q", pod.Name, pj.ClusterAlias())
 			}
-			return client.Delete(pj.ObjectMeta.Name, &metav1.DeleteOptions{})
+
+			err := client.Delete(pj.ObjectMeta.Name, &metav1.DeleteOptions{})
+			c.log.WithField("name", pj.ObjectMeta.Name).Debug("Delete Pod.")
```
Comment (alvaroaleman) [inline prow/plank/controller.go]: What about adding the error as well?

Comment (clarketm) [reply prow/plank/controller.go]: I figured these were already collected so I did not log it. https://github.com/kubernetes/test-infra/blob/c6b4dedc69a097d9cd4d37292514310065e4b3ac/prow/plank/controller.go#L287

Comment (alvaroaleman) [reply prow/plank/controller.go]: okay. Tiny tiny nit then, move the log above the `Delete` call and keep the `return client.Delete`? Right now I intuitively expect the error to be used in the log from the way the code is structured

Comment (clarketm) [reply prow/plank/controller.go]: Updated! That is much cleaner.

**Comment by clarketm** [comment]
@stevekuznetsov @cjwagner - can I please get a quick approval for adding debug-level logging back to `plank` for CRUD operations.

**Pull request merged**